### DISCUSSION TITLE
Support comics on Lychee

### DISCUSSION THREAD

**Discussion by OrcVole** [LycheeOrg/Lychee#2293] [Ideas]
### Enhancement

Comics and cartoons are usually composed of several frames which need to be viewed in order. Sometimes the frames are of different sizes and proportions and need to be tiled together in such a way that the frames are still viewed in sequence.

Lychee already has some support for tiling and can arrange various shaped images but perhaps this can be improved.

<img width="213" alt="brave_9NKieqnhTS" src="https://github.com/LycheeOrg/Lychee/assets/68787520/aa8c32d9-30c8-4e2b-9046-e3802ba1db24">
<img width="211" alt="ComicLife3_cEyYbMBqpc" src="https://github.com/LycheeOrg/Lychee/assets/68787520/9d383210-ef7d-400a-910f-529c76f95d86">
<img width="213" alt="ComicLife3_XF7jE6W6i0" src="https://github.com/LycheeOrg/Lychee/assets/68787520/e65758cb-0924-4ede-9ac6-5e216ddd7275">
<img width="212" alt="ComicLife3_0eDhZAyC8k" src="https://github.com/LycheeOrg/Lychee/assets/68787520/39a97f75-596b-49d7-9a7c-2eb785356812">
<img width="215" alt="ComicLife3_aGC10jQ714" src="https://github.com/LycheeOrg/Lychee/assets/68787520/12b321cd-1ca9-47f8-8e47-a683aae8da52">
<img width="213" alt="ComicLife3_k1z3c70E95" src="https://github.com/LycheeOrg/Lychee/assets/68787520/cdc38aeb-8954-4339-be91-7d307ffd73de">
<img width="215" alt="ComicLife3_kOb8YVzBj7" src="https://github.com/LycheeOrg/Lychee/assets/68787520/bf56a8e0-d5f0-4cec-964c-590aa8212a89">
<img width="215" alt="ComicLife3_yw5QLy8QE7" src="https://github.com/LycheeOrg/Lychee/assets/68787520/d3f02ac1-3e35-4623-a169-b29ea87ede03">

**Comment by ildyria** [comment]
I am not sure what you mean.

Also how would Lychee know which panel to place where? We only have the dimension of the image.

**Reply by OrcVole** [reply]
Thanks for asking. For example, after the panels/images have been arrayed, the user could drag/drop or switch around which image sits where. 

The is some Free Software already for packing/tessellating shapes. 

 Masonry:
https://masonry.desandro.com/
Packery:
http://packery.metafizzy.co/
Isotope:
https://isotope.metafizzy.co/
Dynamic Grid:
https://getuikit.com/v2/docs/grid-js.html
